lightning/src/ln/onion_utils.rs

```diff
@@ -404,7 +405,15 @@ pub(super) fn process_onion_failure<T: secp256k1::Signing, L: Deref>(secp_ctx: &
 						else if error_code & UPDATE == UPDATE {
 							if let Some(update_len_slice) = err_packet.failuremsg.get(debug_field_size+2..debug_field_size+4) {
 								let update_len = u16::from_be_bytes(update_len_slice.try_into().expect("len is 2")) as usize;
-								if let Some(update_slice) = err_packet.failuremsg.get(debug_field_size + 4..debug_field_size + 4 + update_len) {
+								if let Some(mut update_slice) = err_packet.failuremsg.get(debug_field_size + 4..debug_field_size + 4 + update_len) {
+									// Historically, the BOLTs were unclear if the message type
+									// bytes should be included here or not. The BOLTs have now
+									// been updated to indicate that they *are* included, but many
+									// nodes still send messages without the type bytes, so we
+									// support both here.
```

Comment:
I see that supporting both is the way to go for the time being. However, could this kind of permissiveness not also become a source of error, e.g., for channel updates/signatures starting by chance with `[0x01, 0x02]`? Maybe I should add a TODO to remove this backwards compatibility sometime in the (hopefully near) future, when everyone has upgraded?

Reply:
Maybe I'm reading the spec wrong, but isn't it starting with `signature`? (cf. https://github.com/lightning/bolts/blob/master/07-routing-gossip.md#the-channel_update-message)

Reply:
Yeah, if I understand the spec correctly, in the worst case we would temporarily disable the channel if the channel update cannot be read and would restore it as soon as we get another update for it.

Since all implementations but LND seem to encode the type already, and they are also working on it (cf. https://github.com/lightningnetwork/lnd/issues/6461) there should not many non-prefixed channel updates transmitted (hopefully) soon (TM). So I agree that the chances can probably be neglected for the time being, I'll however add a TODO and a TRACE log that lets us monitor how often we still see it..

Reply:
> Are you thinking if we accidentally read something bogus? 
Well, at least it could happen. However, you're probably right that the chances for actually receiving a corresponding update that is then also read in a weird way are pretty slim. So +1 for leaving it as is now and removing the error source ASAP.